games/devtest/mods/util_commands/init.lua

```diff
@@ -1,3 +1,40 @@
+if core.is_singleplayer() then
+	local function format_result(success, ...)
+		if success then
+			local res = {}
+			for i = 1, select("#", ...) do
+				local v = select(i, ...)
+				table.insert(res, dump(v))
+			end
+			if #res == 0 then
+				return true, "No return values."
+			end
+			return true, "Return values: " .. table.concat(res, ",\n")
+		end
+		return false, "Error: " .. tostring((...))
+	end
+
+	core.register_chatcommand("lua", {
+		params = "<code>",
+		description = "Execute Lua code (singleplayer-only)",
+		func = function(name, param)
+			local func = load("return " .. param)
+			if not func then
+				local err
+				func, err = load(param)
+				if not func then
+					return false, "Syntax error: " .. err
+				end
+			end
+			setfenv(func, setmetatable({
+				name = name,
+				player = core.get_player_by_name(name),
+			}, {__index = _G}))
```

Comment:
Here's a few more functions I'd like to have: https://github.com/Desour/luacmd/blob/be2b768df585dc0456d8fef71ea788ea97d41b33/PlayerEnv.lua#L8-L34

* `me`, `myname`: Redundant. But I don't think redundant functions hurt. It gives compatibility with luacmd (and its forks).
* `here`: Quite useful imo.
* `print`: Overridden, so you get nice output not only in terminal, but also in chat.
* `branded`: Useful in devtest, if you want to do something with objects.

(Yes, I know the code of the luacmd mod is ugly.)

Reply:
sure